src/helpers/figures/charts/chart_factory.ts

```diff
@@ -168,21 +169,22 @@ export function getSmartChartDefinition(zone: Zone, getters: Getters): ChartDefi
   // Only display legend for several datasets.
   const newLegendPos = dataSetZone.right === dataSetZone.left ? "none" : "top";
 
-  const labelRange = labelRangeXc ? getters.getRangeFromSheetXC(sheetId, labelRangeXc) : undefined;
-  if (canChartParseLabels(labelRange, getters)) {
-    return {
-      title,
-      dataSets,
-      labelsAsText: false,
-      stacked: false,
-      aggregated: false,
-      cumulative: false,
-      labelRange: labelRangeXc,
-      type: "line",
-      dataSetsHaveTitle,
-      verticalAxisPosition: "left",
-      legendPosition: newLegendPos,
-    };
+  const lineChartDefinition: LineChartDefinition = {
+    title,
+    dataSets,
+    labelsAsText: false,
+    stacked: false,
+    aggregated: false,
+    cumulative: false,
+    labelRange: labelRangeXc,
+    type: "line",
+    dataSetsHaveTitle,
+    verticalAxisPosition: "left",
+    legendPosition: newLegendPos,
+  };
+  const chart = new LineChart(lineChartDefinition, sheetId, getters);
```

Comment:
I personnaly find it useless to create a LineChart here just so that we have the correct object to pass to `canChartPaseLabels`, while this function will only use labelRange, dataSets[0] and dataSetsHaveTitle from the full chart.
Maybe we can create a temporary object if you really want to have only one argument, but I would go with 4 distincs arguments (`labelRange`, `dataSet = dataSets[0]`, `dataSetsHaveTitle` and `getters`)

Reply:
Moreover, the only reason you need these new arguments is for the "shouldRemoveFirstLabel" function calls. So maybe we can call it here before calling canChartParseLabels, and then call the later with `labelRange`, `removeFirstLabel` and `getters` ?

Reply:
We don't have actual `DataSet` objects here, only zones/XCs. Se we'd have to call `createDataSets` first. And I'm not a big fan of calling 3 helpers just to get the parameters to call another helpers because 1) it's annoying and 2) there's no guarantee that we call the same helpers/in the same way that the actual chart does. So it's way easier for errors to slip by. I'd much prefer just create an actual LineChart.